storage/remote/otlptranslator/prometheus/normalize_name.go

```diff
@@ -88,45 +88,51 @@ var perUnitMap = map[string]string{
 // See rules at https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels,
 // https://prometheus.io/docs/practices/naming/#metric-and-label-naming
 // and https://github.com/open-telemetry/opentelemetry-specification/blob/v1.38.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus.
-func BuildCompliantName(metric pmetric.Metric, namespace string, addMetricSuffixes bool) string {
+func BuildCompliantName(metric pmetric.Metric, namespace string, addMetricSuffixes, allowUTF8 bool) string {
 	// Full normalization following standard Prometheus naming conventions
 	if addMetricSuffixes {
-		return normalizeName(metric, namespace)
+		return normalizeName(metric, namespace, allowUTF8)
 	}
 
-	// Regexp for metric name characters that should be replaced with _.
-	invalidMetricCharRE := regexp.MustCompile(`[^a-zA-Z0-9:_]`)
-
-	// Simple case (no full normalization, no units, etc.).
-	metricName := strings.Join(strings.FieldsFunc(metric.Name(), func(r rune) bool {
-		return invalidMetricCharRE.MatchString(string(r))
-	}), "_")
+	var metricName string
+	if !allowUTF8 {
+		// Regexp for metric name characters that should be replaced with _.
+		invalidMetricCharRE := regexp.MustCompile(`[^a-zA-Z0-9:_]`)
+
+		// Simple case (no full normalization, no units, etc.).
+		metricName = strings.Join(strings.FieldsFunc(metric.Name(), func(r rune) bool {
+			return invalidMetricCharRE.MatchString(string(r))
+		}), "_")
+	} else {
+		metricName = metric.Name()
+	}
 
 	// Namespace?
 	if namespace != "" {
 		return namespace + "_" + metricName
 	}
 
-	// Metric name starts with a digit? Prefix it with an underscore.
-	if metricName != "" && unicode.IsDigit(rune(metricName[0])) {
+	// Metric name starts with a digit and utf8 not allowed? Prefix it with an underscore.
```

Comment:
@jesusvazquez do you know if this change is correct?

Reply:
I have no clue 😱